### PR TITLE
Setup prince

### PR DESCRIPTION
**Pull request by PhiAgent**
Qtns & Ans routes complete @miguelmar21 , @AdamRakla

### CODE REVIEW

File: server/QuestionsAndAnswers/answers.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var routes = require('express').Router();
+var axios = require('axios');
+var url = require('url');
+var config = require('../../config');
+var endPoint = 'https://app-hrsei-api.herokuapp.com/api/fec2/:hr-atx/qa/answers'
+
+var auth = { headers: { 'Authorization': `${config}` }, params: { product_id: 1 } };
+
+routes.put(`/:answer_id/*`, (req, res) => {
+  var path = url.parse(req.url, true).path.split('/').at(-1);
```
Comment (nickfreemandesign) [inline server/QuestionsAndAnswers/answers.js:10]: can be `const`. also, what happens if there is nothing to split `.at(-1)`?

File: server/QuestionsAndAnswers/answers.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  axios.put(endPoint + `/${req.params.answer_id}/${path}`, { answer_id: req.params.answer_id }, auth)
```
Comment (nickfreemandesign) [inline server/QuestionsAndAnswers/answers.js:11]: you may want to drive consistency here with the same convention for string interpolation/concatenation:

```node.js
const [ PATH ] = new url(url).path.split("/")
const ENDPOINT = "https://app-hrsei-api.herokuapp.com/api/fec2/:hr-atx/qa/answers"
const { 
  params: { answer_id: ANSWER_ID },
  url
} = req

const ENDPOINT_WITH_PATH = `${ENDPOINT}/${ANSWER_ID}/${PATH}`

axios.put(
  ENDPOINT_WITH_PATH,
  { answer_id: ANSWER_ID },
  auth
)
```

File: server/QuestionsAndAnswers/questions.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+var routes = require('express').Router();
+var axios = require('axios');
+var url = require('url');
+var config = require('../../config');
```
Comment (nickfreemandesign) [inline server/QuestionsAndAnswers/questions.js:4]: based on your usage, this isn't a `config` but this is an `authorization` that comes from your `config`. the goal is to arrive at a place where your variables all align with the purpose they serve, without any ambiguity

File: server/QuestionsAndAnswers/questions.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+var endPoint = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-atx/qa/questions';
+
+var auth = { headers: { 'Authorization': `${config}` }};
+
+routes.get('/', (req, res) => {
+  axios.get(endPoint, { headers: { 'Authorization': `${config}` }, params: { product_id: req.query.product_id } })
```
Comment (nickfreemandesign) [inline server/QuestionsAndAnswers/questions.js:10]: you have `auth` defined on line `7`, may make sense to use that here in line `10`

File: server/QuestionsAndAnswers/questions.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  axios.get(endPoint + `/${req.params.question_id}/answers`, auth)
+    .then(response => res.status(200).send(response.data))
+    .catch(err => {
+      console.error(err);
+      res.status(400).end();
+    })
+})
+
+routes.post(`/:product_id/question`, (req, res) => {
+  let body = req.body.body, name = req.body.name, email = req.body.email, product_id = req.body.product_id;
```
Comment (nickfreemandesign) [inline server/QuestionsAndAnswers/questions.js:28]: this is prime for destructure!

```node.js
const {
  body,
  name,
  email,
  product_id: productId
} = req.body
```

File: server/QuestionsAndAnswers/questions.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  axios.post(endPoint + `/${req.params.question_id}/answers`, { body: req.body.body, name: req.body.name, email: req.body.email, photos: req.body.photos }, { headers: { 'Authorization': `${config}` }, params: { question_id: req.params.question_id } })
+    .then(response => res.status(200).send(response.statusText))
+    .catch(err => {
+      console.error(err);
+      res.status(400).end();
+    })
+})
+
+routes.put(`/:question_id/*`, (req, res) => {
+  var path = url.parse(req.url, true).path.split('/').at(-1);
```
Comment (nickfreemandesign) [inline server/QuestionsAndAnswers/questions.js:47]: if you plan to use this convention a number of times, you're best to factor it out to a utility function `getFirstPath` or something. This way you can error handle and do other things to make this a more reliable maneuver.

**Review by nickfreemandesign — COMMENTED** [review]
@PhiAgent thanks for sharing this PR with me! I can see you are starting to build a coding paradigm in this PR that suggests how new routes will be made. Ultimately, this will result in "boilerplate" code, which is ok (and even good!). When you start to notice yourself writing boilerplate, ask what are some ways you can reduce the code that needs to be written every time you make a route? every line of boilerplate code reduced will save on mistakes made during development

File: server/QuestionsAndAnswers/answers.js (hunk 1 above)
Comment (PhiAgent) [reply server/QuestionsAndAnswers/answers.js:10]: Yes const is a better key word. 
Hmm, it seems i didn't account for the possibility of there being nothing to split. I have to account for that.

File: server/QuestionsAndAnswers/questions.js (hunk 3 above)
Comment (PhiAgent) [reply server/QuestionsAndAnswers/questions.js:4]: Yes, i just looked it up. it's actually my git token. I think a variable name like gitToken will better align with the purpose.

File: server/QuestionsAndAnswers/questions.js (hunk 4 above)
Comment (PhiAgent) [reply server/QuestionsAndAnswers/questions.js:10]: Oh yes, i learned about the use of the spread operator on objects after this part. I'll apply it hear like :
{...auth, params:{}}

File: server/QuestionsAndAnswers/questions.js (hunk 5 above)
Comment (PhiAgent) [reply server/QuestionsAndAnswers/questions.js:28]: Oh wow, that's so true. I need to get more comfortable with destructure

File: server/QuestionsAndAnswers/questions.js (hunk 6 above)
Comment (PhiAgent) [reply server/QuestionsAndAnswers/questions.js:47]: I actually felt bad repeating this line a number of times. You're right, I should isolate it. I'm gonna keep that in mind in future projects.

File: server/QuestionsAndAnswers/answers.js (hunk 2 above)
Comment (PhiAgent) [reply server/QuestionsAndAnswers/answers.js:11]: Hmm, the [ PATH ] syntax, seems unfamiliar, reading up on it now. And the consistency is nice. i prefer the way you put it.

**Pull request merged**